### PR TITLE
Features/dbup landing page changes

### PR DESCRIPTION
**Pull request by ssrajan**
1. Recent changes for the Landing page
2. New page modified as per the requirement 
   a ) new combo box for Production Action Required is added
  b ) As per the selection , the SNC and ENC steps are added into the workflow
3. Recent fixes such as  'Double' clicking of Create New button, tab order fix are fixed here also.

### CODE REVIEW

File: src/Databases/DbUpWorkflowDatabase/Tables/ProductAction.sql [hunk 1]
```diff
@@ -0,0 +1,6 @@
+﻿CREATE TABLE [dbo].[ProductAction]
+(
+	[ProductActionId] INT NOT NULL PRIMARY KEY, 
+    [Name] NCHAR(50) NOT NULL,
+	CONSTRAINT [AK_Production_Action_Name] UNIQUE ([Name])
```
Comment (UKHOGreg) [inline src/Databases/DbUpWorkflowDatabase/Tables/ProductAction.sql:5]:
```suggestion
	CONSTRAINT [AK_ProductAction_Name] UNIQUE ([Name])
```

File: src/DbUpdatePortal/Helpers/PageValidationHelper.cs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+                isValid = false;
+            }
+            if (string.IsNullOrEmpty(updateType))
+            {
+                validationErrorMessages.Add("Task Information: Update Type cannot be empty");
+                isValid = false;
+            }
+            if (string.IsNullOrEmpty(productAction))
+            {
+                validationErrorMessages.Add("Task Information : Product Action Required cannot be empty");
```
Comment (UKHOGreg) [inline src/DbUpdatePortal/Helpers/PageValidationHelper.cs:41]:
```suggestion
                validationErrorMessages.Add("Task Information: Product Action Required cannot be empty");
```

File: src/DbUpdatePortal/Helpers/StageTypeFactory.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            //        taskStageTypes.Add(tsTypes.Single(t => t.TaskStageTypeId == (int)DbUpdateTaskStageType.SNC));
+            //        break;
+            //    case DbUpdateProductAction.ENC:
+            //        taskStageTypes.Add(tsTypes.Single(t => t.TaskStageTypeId == (int)DbUpdateTaskStageType.ENC));
+            //        break;
+            //    default: // option "SNC & ENC" is selected
+            //        taskStageTypes.AddRange(tsTypes.Where(t =>
+            //            t.TaskStageTypeId > (int)DbUpdateTaskStageType.Verification_Rework));
+            //        break;
+            //}
```
Comment (UKHOGreg) [inline src/DbUpdatePortal/Helpers/StageTypeFactory.cs:43]: Comment intentional?

File: src/DbUpdatePortal/Enums/DbUpdateProductAction.cs [hunk 4]
```diff
@@ -0,0 +1,10 @@
+﻿namespace DbUpdatePortal.Enums
+{
+    public enum DbUpdateProductAction
+    {
+        None = 1,
+        ENC = 2,
+        SNC = 3,
+        Both = 4
+    }
+}
```
Comment (SamirHassoun) [inline src/DbUpdatePortal/Enums/DbUpdateProductAction.cs:10]: Aren't you using a DB table ProductAction for this?

File: src/Databases/DbUpWorkflowDatabase/Tables/ProductAction.sql (hunk 1 above)
Comment (ssrajan) [reply src/Databases/DbUpWorkflowDatabase/Tables/ProductAction.sql:5]: Done

File: src/DbUpdatePortal/Helpers/PageValidationHelper.cs (hunk 2 above)
Comment (ssrajan) [reply src/DbUpdatePortal/Helpers/PageValidationHelper.cs]: Done

File: src/DbUpdatePortal/Helpers/StageTypeFactory.cs (hunk 3 above)
Comment (ssrajan) [reply src/DbUpdatePortal/Helpers/StageTypeFactory.cs]: Removed the comments. For Db Update, we may not need StageTypeFactory. As of now it brings all the types. I'll remove the factory during workflow page

File: src/DbUpdatePortal/Enums/DbUpdateProductAction.cs (hunk 4 above)
Comment (ssrajan) [reply src/DbUpdatePortal/Enums/DbUpdateProductAction.cs:10]: Yes, Combobox is populated from the db table. This enum is used to map the value selected and then decide the action on it.

**Comment by ssrajan** [comment]
> Few minor things!
> 
> Also, there are a few instances where `userPrinciple` is used, such as:
> `LogContext.PushProperty("UserPrincipalName", userPrinciple);`
> I would expect to see `userPrincipal` instead.
> 
> 👍

File: src/DbUpdatePortal/Enums/DbUpdateProductAction.cs (hunk 4 above)
Comment (SamirHassoun) [reply src/DbUpdatePortal/Enums/DbUpdateProductAction.cs:10]: this is a bit risky; can't you use the DB table for that?

File: src/Databases/DbUpWorkflowDatabase/Tables/ProductAction.sql (hunk 1 above)
Comment (SamirHassoun) [reply src/Databases/DbUpWorkflowDatabase/Tables/ProductAction.sql]: Am I missing something? its still says "CONSTRAINT [AK_Product_Action_Name] UNIQUE ([Name])" while @UKHOGreg was suggesting "CONSTRAINT [AK_ProductAction_Name] UNIQUE ([Name])"

File: src/DbUpdatePortal/Pages/NewTask.cshtml.cs [hunk 5]
```diff
@@ -298,6 +269,19 @@ private void SetUpdateTypes()
...
+                throw new FileNotFoundException(@"Data\ProductActions.json");
+
+            var jsonString = System.IO.File.ReadAllText(@"Data\ProductActions.json");
+
+            var productActions = JsonConvert.DeserializeObject<IEnumerable<ProductAction>>(jsonString)
+                .Select(u => u.Name);
+
+            ProductActions = new SelectList(productActions);
+        }
+
```
Comment (SamirHassoun) [inline src/DbUpdatePortal/Pages/NewTask.cshtml.cs:284]: Are planning to populate from DB table ProductAction later?

File: src/Databases/DbUpWorkflowDatabase/Tables/ProductAction.sql (hunk 1 above)
Comment (ssrajan) [reply src/Databases/DbUpWorkflowDatabase/Tables/ProductAction.sql]: It was ProductionAction, I changed to Product_Action. Let us leave it there  :)

File: src/DbUpdatePortal/Enums/DbUpdateProductAction.cs (hunk 4 above)
Comment (ssrajan) [reply src/DbUpdatePortal/Enums/DbUpdateProductAction.cs:10]: It's always used like this in NCNE. Let's have a review later and find a better way to do this. Thanks

File: src/DbUpdatePortal/Pages/NewTask.cshtml.cs (hunk 5 above)
Comment (ssrajan) [reply src/DbUpdatePortal/Pages/NewTask.cshtml.cs]: all combobox population is done using DB now. Changed the NCNE new page also

File: src/Databases/DbUpWorkflowDatabase/Tables/ProductAction.sql [hunk 6]
```diff
@@ -0,0 +1,6 @@
+﻿CREATE TABLE [dbo].[ProductAction]
+(
+	[ProductActionId] INT NOT NULL PRIMARY KEY, 
+    [Name] NCHAR(50) NOT NULL,
```
Comment (Rossall82) [inline src/Databases/DbUpWorkflowDatabase/Tables/ProductAction.sql:4]: Can we use nvarchar here for consistency?

File: src/Databases/DbUpWorkflowDatabase/Tables/ProductAction.sql [hunk 7]
```diff
@@ -0,0 +1,6 @@
...
+	CONSTRAINT [AK_Product_Action_Name] UNIQUE ([Name])
```
Comment (Rossall82) [inline src/Databases/DbUpWorkflowDatabase/Tables/ProductAction.sql:5]: I think as others said, the convention for a constraint would be AK_TABLENAME_COLUMNNAME, so we should go for AK_ProductAction_Name here.

File: src/Common.Helpers/DbUpdateWorkflowDatabaseSeeder.cs [hunk 8]
```diff
@@ -31,7 +31,8 @@ public ICanSaveChanges PopulateTables()
             PopulateTaskStageType();
             PopulateChartingArea();
             PopulateUpdateType();
-            PopulateTaskInfo();
+            PopulateProductAction();
+            //PopulateTaskInfo();
```
Comment (Rossall82) [inline src/Common.Helpers/DbUpdateWorkflowDatabaseSeeder.cs:35]: Do we want this commented out?

File: src/Databases/DbUpWorkflowDatabase/Tables/ProductAction.sql (hunk 6 above)
Comment (ssrajan) [reply src/Databases/DbUpWorkflowDatabase/Tables/ProductAction.sql:4]: Done

File: src/Databases/DbUpWorkflowDatabase/Tables/ProductAction.sql (hunk 7 above)
Comment (ssrajan) [reply src/Databases/DbUpWorkflowDatabase/Tables/ProductAction.sql]: Done

File: src/Common.Helpers/DbUpdateWorkflowDatabaseSeeder.cs (hunk 8 above)
Comment (ssrajan) [reply src/Common.Helpers/DbUpdateWorkflowDatabaseSeeder.cs]: Removed the comment

**Pull request merged**